webclient/src/components/Home/SavedDatasetList.tsx

```diff
@@ -1,10 +1,50 @@
+import { Icon } from '@chakra-ui/icons';
 import { Table, Tbody, Td, Th, Thead, Tr } from '@chakra-ui/table';
-import { FC, useEffect, useRef, useState } from 'react';
+import { Avatar, Box, Flex, Heading, Text } from '@chakra-ui/react';
+import { FC, useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
+import { MdDeleteOutline, MdModeEditOutline, MdOutlineMap, MdOutlineTextSnippet } from "react-icons/md";
 import { useRecoilValue } from 'recoil';
 import { useRemoveDatasetFromLocalstorage } from '../../hooks/useDataset';
 import { datasetListSelector } from '../../stores/dataset';
 import { OstButton } from '../Elements/OstButton';
+import { OstNavLink } from '../Elements/OstLink';
+
+const RemoveAllButton: FC = () => {
+  const datasets = useRecoilValue(datasetListSelector);
+  // HACK: ループ処理で各datasetにuseRemoveDatasetFromLocalstorageを定義することができないため、1要素ずつ削除
+  // (関数コンポーネント内のフックの個数が動的に変化するとReactがクラッシュするため https://ja.reactjs.org/docs/hooks-rules.html)
+  const [deleting, setDeleting] = useState(false);
+  // 先頭のdatasetを削除する関数
+  const { removeFromLocalstorage } = useRemoveDatasetFromLocalstorage({datasetUid: String(datasets[0]?.uid)});
```

Comment:
単純にdatasetsの各要素に対しfor文で `useRemoveDatasetFromLocalstorage` を行うと、要素数によってrecoilのフックの個数が変わってしまい削除時にクラッシュしてしまいました。

```
Uncaught Error: Rendered fewer hooks than expected. This may be caused by an accidental early return statement.
```

あまりお行儀のよい解決策ではないので、もっと良い方法があればそちらに置き換えたいと思っています。

Reply:
なるほどです...！私も手元で削除ボタンの挙動がおかしくなるな...と思ってました。
これ結構クリティカルだと思うので、早急に要検討ですね 👀